src/App.js

```diff
@@ -1,23 +1,133 @@
-import logo from './logo.svg';
-import './App.css';
+import "./App.css";
+
+import Header from "./components/Header";
+import DisplayAlgo from "./components/DisplayAlgo";
+import DisplayAlgoLine from "./components/DisplayAlgoLine";
+import Footer from "./components/Footer";
+import { useState, useEffect } from "react";
 
 function App() {
+  const numRow = Math.floor(window.innerWidth / 11);
+  const [runningAlgorithm, setRunningAlgorithm] = useState(false);
+  const [rows, setRows] = useState([]);
+  const [arrayAccessCounter, setArrayAccessCounter] = useState(0);
+  const [comparisonCounter, setComparisonCounter] = useState(0);
+  //_____________________Functions________________________________________
+
+  //randomize graph
+  const randomize = () => {
+    const tempArr = [];
+    for (var i = 0; i < numRow; i++) {
+      tempArr.push({
+        height: Math.floor(Math.random() * 500),
+        id: i,
+      });
+    }
+    setRows((rows) => tempArr);
+  };
+
+  //setStates running program
+  const changeRunningAlgorithm = (running) => {
+    if (running) {
+      setRunningAlgorithm(true);
+    } else {
+      setRunningAlgorithm(false);
+    }
+  };
```

Comment:
This whole function is probably unnecessary abstraction. First, the `if/else` on it is simply direct argument passing:

```suggestion
  const changeRunningAlgorithm = (running) => {
    setRunningAlgorithm(running)
  };
```

Which, if you think further, if both receive the same value and have the same outcome, they are identical:

```suggestion
  const changeRunningAlgorithm = setRunningAlgorithm;
```

Meaning in every use case of the `changeRunningAlgorithm`, you can, and I think should, use `setRunningAlgorithm`.

However, there are many times when a function renaming or abstraction is the right choice. That's specially true when the _intent_ of the code executing the function has another meaning. A good example is when you have code executed upon user events: you could have a `handleClick` function, which executes a `exit` function. The first, is a _functional_ naming: it's bound to it's place of use. The second, is a _semantic_ naming: it describes an action.